### PR TITLE
Update developerGuide.md

### PR DESCRIPTION
**Pull request by gohyongjing**
#### Use Cases

* Remove implementation details ('FL saves the data')
* Remove UI details ('User enters `xxx` into input box')
* Remove non-interaction details ('User sees the updated info')

#### Stats Command
* Add Sequence diagram for `execute` method

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -464,11 +467,12 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 **Use case: UC01- Pair Volunteer and Elderly**
 
+Precondtions: Elderly and Volunteer is already in FL.
+
 **MSS**
 
 1.  User enters the details of elderly and volunteer to be paired into the application.
-2.  FL adds the pair into the database, and feedbacks the successful addition of the pair.
-3.  User see the pair details appear in the pair list.
+2.  FL feedbacks the successful addition of the pair, and shows the new pair.
```
Comment (heeeyi) [inline docs/DeveloperGuide.md:476]: I feel here should show the FL feedback and user-side feedback, so better maintain the original format

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -499,19 +503,18 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 **MSS**
 
 1.  User enters the details of elderly to be added into the application.
-2.  FL adds the elderly into the database, and feedbacks the successful addition of the elderly.
-3.  User see the elderly details appear in the elderly list.
+2.  FL feedbacks the successful addition of the elderly, and shows the new elderly.
```
Comment (heeeyi) [inline docs/DeveloperGuide.md:507]: Same that show both FL and User side reaction

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -521,69 +524,65 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 **MSS**
 
 1.  User enters the details of volunteer to be added into the application.
-2.  FL adds the volunteer into the database, and feedbacks the successful addition of the volunteer.
-3.  User see the volunteer details appear in the volunteer list.
+2.  FL feedbacks the successful addition of the volunteer and shows the new volunteer.
```
Comment (heeeyi) [inline docs/DeveloperGuide.md:527]: all below same

**Review by heeeyi — COMMENTED** [review]
Nice work 👍

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (gohyongjing) [reply docs/DeveloperGuide.md:477]: Resolved in [a108200](https://github.com/AY2223S2-CS2103T-W12-1/tp/pull/294/commits/a1082000ca6232ebe7eed75ff3d81f9e2a2dac2a)

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (gohyongjing) [reply docs/DeveloperGuide.md]: Resolved in [a108200](https://github.com/AY2223S2-CS2103T-W12-1/tp/pull/294/commits/a1082000ca6232ebe7eed75ff3d81f9e2a2dac2a)

**Review by wz2k — APPROVED** [review]
LGTM

**Pull request merged**